File: use-cases/data-kiosk/code/python/src/cancel_query_handler.py
import logging
import os

from src.utils.api_utils import ApiUtils
from src.utils import constants

logger = logging.getLogger()
logger.setLevel(logging.INFO)

"""
{
  "QueryId": "1231413413531"
}
"""


def lambda_handler(event, context):
    # Retrieve request details from the input payload
    query_id = event[constants.QUERY_ID_KEY_NAME]
    region_code = os.environ.get(constants.REGION_CODE_ARN_ENV_VARIABLE)
    refresh_token = os.environ.get(constants.REFRESH_TOKEN_ARN_ENV_VARIABLE)

    # Log the input event for the Lambda
    logger.info(f"CancelQuery Lambda input: {event}")

    try:
        # Create an instance of the ApiUtils class
        api_utils = ApiUtils(refresh_token, region_code, constants.DATA_KIOSK_API_TYPE)

        # Call the cancel_query method to cancel the query
        api_utils.call_datakiosk_api('cancel_query', query_id=query_id)

        return f"Query {query_id} has been canceled successfully!"

    except Exception as e:
        raise Exception("Calling Data Kiosk API failed", e.__dict__)

File: use-cases/data-kiosk/code/java/src/main/java/lambda/CancelQueryHandler.java
package lambda;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.google.gson.Gson;
import io.swagger.client.api.QueriesApi;
import lambda.utils.ApiUtils;

import java.util.Map;

import static lambda.utils.Constants.QUERY_ID_KEY_NAME;
import static lambda.utils.Constants.REFRESH_TOKEN_ARN_ENV_VARIABLE;
import static lambda.utils.Constants.REGION_CODE_ARN_ENV_VARIABLE;

public class CancelQueryHandler implements RequestHandler<Map<String, String>, String> {

    // Sample event input:
    // {
    //  "QueryId": "1231413413531"
    // }

    public String handleRequest(Map<String, String> input, Context context) {
        LambdaLogger logger = context.getLogger();
        logger.log("CancelQuery Lambda input: " + new Gson().toJson(input));

        // Retrieve request details from the input payload
        String queryId = input.get(QUERY_ID_KEY_NAME);
        String regionCode = System.getenv(REGION_CODE_ARN_ENV_VARIABLE);
        String refreshToken = System.getenv(REFRESH_TOKEN_ARN_ENV_VARIABLE);

        try {
            // Initialize the DataKiosk API client
            QueriesApi dataKioskApi = ApiUtils.getDataKioskApi(regionCode, refreshToken);

            // Call the cancelQuery method to create the query
            dataKioskApi.cancelQuery(queryId);

            return String.format("Query %s has been canceled successfully", queryId);
        } catch (Exception e) {
            throw new InternalError("Cancel query failed", e);
        }
    }
}